site/app/controllers/forum/ForumController.php

```diff
@@ -85,23 +94,54 @@ private function checkGoodAttachment($isThread, $thread_id, $file_post){
         } return $imageCheck;
     }
 
+    private function hasGoodCategory($inputCategoryId){
+        if($inputCategoryId < 1)
+            return false;
+        $rows = $this->core->getQueries()->getCategories();
+        foreach($rows as $index => $values){
+            if($values["category_id"] === $inputCategoryId)
+                return true;
+        } return false;
+    }
+
+    public function addNewCategory(){
+        $result = array();
+        if($this->core->getUser()->getGroup() <= 2){
+            if(!empty($_REQUEST["newCategory"])) {
+                $category = $_REQUEST["newCategory"];
+                $newCategoryId = $this->core->getQueries()->addNewCategory($category);
+                $result["new_id"] = $newCategoryId["category_id"];
+            } else {
+                $result["error"] = "No category data submitted";
+            }
+        } else {
+            $result["error"] = "You do not have permissions to do that.";
+        } 
+        $this->core->getOutput()->renderJson($result);
+        return $result;
+    }
+
     //CODE WILL BE CONSOLIDATED IN FUTURE
 
     public function publishThread(){
         $title = $_POST["title"];
         $thread_content = str_replace("\r", "", $_POST["thread_content"]);
         $anon = (isset($_POST["Anon"]) && $_POST["Anon"] == "Anon") ? 1 : 0;
         $announcment = (isset($_POST["Announcement"]) && $_POST["Announcement"] == "Announcement" && $this->core->getUser()->getGroup() < 3) ? 1 : 0 ;
+        $category_id = $_POST["cat"];
```

Comment:
You should just cast this to an int immediately and not do a `is_numeric()` and `empty()` check. Just cast it and use `hasGoodCategory`.